apps/omg/lib/omg/state/core.ex

```diff
@@ -161,16 +183,25 @@ defmodule OMG.State.Core do
   """
   @spec exec(state :: t(), tx :: Transaction.Recovered.t(), fees :: Fees.optional_fee_t()) ::
           {:ok, {Transaction.tx_hash(), pos_integer, non_neg_integer}, t()}
-          | {{:error, Validator.exec_error()}, t()}
+          | {{:error, Validator.process_error()}, t()}
   def exec(%Core{} = state, %Transaction.Recovered{} = tx, fees) do
     tx_hash = Transaction.raw_txhash(tx)
 
-    case Validator.can_apply_spend(state, tx, fees) do
-      true ->
+    case Validator.can_process(state, tx, fees) do
+      {:ok, :apply_spend, fees_paid} ->
```

Comment:
I'm not a fan of the Validator telling State.Core how to apply the tx like this. But I see where this is coming from, hmmm...